python/hail/java.py

```diff
@@ -119,3 +128,42 @@ def handle_py4j(func, *args, **kwargs):
         else:
             raise e
     return r
+
+
+class LoggingTCPHandler(SocketServer.StreamRequestHandler):
+    def handle(self):
+        for line in self.rfile:
+            sys.stderr.write(line)
+
+
+class SimpleServer(SocketServer.ThreadingMixIn, SocketServer.TCPServer):
+    daemon_threads = True
+    allow_reuse_address = True
+
+    def __init__(self, server_address, RequestHandlerClass):
+        SocketServer.TCPServer.__init__(self, server_address, RequestHandlerClass)
+
+
+def connect_logger(host, port):
+    server = None
+    tries = 0
+    max_tries = 25
+    while not server:
+        try:
+            server = SimpleServer((host, port), LoggingTCPHandler)
```

Comment:
You should put make this retry loop infinite, add a 1s sleep on failure, and put both the server creation and the call to serve_forever in the loop.